### PR TITLE
add privileged parameter to TaskDefinition

### PR DESCRIPTION
**Pull request by mapsam**
This adds a watchbot parameter `privileged` that provides an opportunity to run the container with elevated credentials.

cc @emilymdubois @rclark @springmeyer

### CODE REVIEW

**Comment by rclark** [comment]
Do you also need to consider ReadOnlyRootFileSystem on the task definition? https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions.html#cfn-ecs-taskdefinition-containerdefinition-readonlyrootfilesystem?

**Comment by mapsam** [comment]
Good point @rclark. @springmeyer brings up another point that we may need to edit `ulimit` info as well: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions.html#cfn-ecs-taskdefinition-containerdefinition-ulimits - if the goal is to edit core files.

**Comment by mapsam** [comment]
Confirmed this works - ready for a final review @rclark @emilymdubois!

File: lib/template.js [hunk 1]
```diff
@@ -364,6 +367,8 @@ module.exports = function(options) {
           Image: cf.join([cf.accountId, '.dkr.ecr.', cf.findInMap('EcrRegion', cf.region, 'Region'), '.amazonaws.com/', options.service, ':', options.serviceVersion]),
           Memory: options.reservation.memory,
           Cpu: options.reservation.cpu,
+          Privileged: options.privileged,
+          ReadonlyRootFilesystem: options.privileged ? false : true,
```
Comment (rclark) [inline lib/template.js]: Don't actually need the `? false : true` here, right?

Comment (mapsam) [inline lib/template.js]: I wasn't sure, so I played it safe. Think it should just be `false` all the time?

Comment (rclark) [inline lib/template.js]: I mean that `options.privileged` is already either `true` or `false`. Do you need another conditional?

Comment (mapsam) [inline lib/template.js]: Ah, I see! Yeah, this can just be `!options.privileged` if it's necessary.

**Comment by mapsam** [comment]
@rclark confirmed this works just fine without the ReadOnlyRootFilesystem addition.

**Comment by rclark** [comment]
Thanks @mapsam, looks great.

**Comment by mapsam** [comment]
Well, apparently `git rebase master` and `git push origin <branch>` isn't the right way to do things from your local checkout. I squashed the duplicate history on the merge, so we should be fine. Sorry for the cruft!

**Pull request merged**